codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpProtocolUnitTestRequestGenerator.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ *
+ */
+
+package software.amazon.smithy.go.codegen.integration;
+
+import java.util.List;
+import java.util.logging.Logger;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoDependency;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.protocoltests.traits.HttpRequestTestCase;
+
+public class HttpProtocolUnitTestRequestGenerator extends HttpProtocolUnitTestGenerator<HttpRequestTestCase> {
+    private static final Logger LOGGER = Logger.getLogger(HttpProtocolUnitTestRequestGenerator.class.getName());
+
+    public static class Builder {
```

Comment:
It doesn't look like you're using this as a builder, but rather as a provider? Can you explain a little about the design here?

Reply:
Mainly the idea here was to provide a way to for a builder to be provided to the caller so that the caller can create context specific HttpProtocolUnitTestRequestGenerator  value. This design was used to allow the concrete implementations of `HttpProtocolUnitTestRequestGenerator` to be extended and custom implementations to be used if needed. This also allows as dependency injects in the ProtocolGenerator in aws-sdk-go-v2's codegen, https://github.com/aws/aws-sdk-go-v2/pull/577

Reply:
Will update PR without that stuff,

Reply:
Given the context, I'd recommend moving this to a more standard Java `Builder` pattern. This would have individual methods for setting the components, [like on this class](https://github.com/awslabs/smithy-typescript/blob/0b83c74fd13321ef1ad008185322f9dc5fcd9b75/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/RuntimeClientPlugin.java#L289).

Reply:
Wouldn't that implies that the fields were optional? Whereas a single method specifies all the required members.

Reply:
If you have fields that are not optional can't you use the SmithyBuilder interface to assert what is required?